verify/merkle.go

```diff
@@ -7,9 +7,14 @@ import (
 	"github.com/ethereum/go-ethereum/crypto"
 )
 
-// ProcessInclusionProof processes the Merkle root proof
+// ProcessInclusionProof computes the merkle root hash based on the provided leaf and proof, returning the result.
+// An error is returned if the proof param is malformed.
+//
+// NOTE: this method returning a nil error does NOT indicate that the proof is valid. Rather, it merely indicates that
+// the proof was well-formed. The hash returned by this method must be compared to the claimed root hash, to
+// determine if the proof is valid.
 func ProcessInclusionProof(proof []byte, leaf common.Hash, index uint64) (common.Hash, error) {
```

Comment:
Can we add a comment explaining that index is the index of the leaf. Also is it leafIndex? or index in the tree? does it start at 0? etc

Reply:
Clarification added in [26d68ca](https://github.com/Layr-Labs/eigenda-proxy/pull/219/commits/26d68ca1b1c7bfe12c678b5c481b1865fd567bbd)